### PR TITLE
Scilla call tests

### PR DESCRIPTION
**Pull request by bzawisto**
## Description
Luckily, the issue with invisible scilla updates to evm contract withing single frame turned out to be rather straightforward to fix. (See ScillaCallComplex.sol).
There are also many new tests added.

## Backward Compatibility

- [ ] This is not a breaking change
- [ ] This is a breaking change

### Implementation

- [x] **ready for review**

### Integration Test (Core Team)

- [ ] local machine test

- [ ] small-scale cloud test

### CODE REVIEW

**Review by Steve-White-UK — COMMENTED** [review]
Looking Good - Executing tests as we speak, expect to approve later if passes successfully.

**Review by Steve-White-UK — APPROVED** [review]
Passes the tests nicely, should probably get an rrw stamp but LGTM

File: tests/EvmAcceptanceTests/contracts/rfc75/ScillaCallComplex.sol
Comment (rrw-zilliqa) [inline tests/EvmAcceptanceTests/contracts/rfc75/ScillaCallComplex.sol:74]: Add some negative tests here - what happens if success == false? If the txn can succeed when success=false, test this.

Test booleans, ints, ByStr, Address, List (and update RFC075 to indicate that maps are implemented)

File: tests/EvmAcceptanceTests/contracts/scilla/rfc75/ScillaCallComplex.scilla
Comment (rrw-zilliqa) [inline tests/EvmAcceptanceTests/contracts/scilla/rfc75/ScillaCallComplex.scilla:27]: use different values for each of simpleMap, nestedMap, and value so that if you read the wrong one in the chain code, the test fails.

**Review by rrw-zilliqa — APPROVED** [review]
LGTM on a first look, but worth addressing the comments in another PR - thanks!

**Pull request merged**
